pkg/controller/servicebinding/retriever.go

```diff
@@ -44,7 +45,10 @@ func buildServiceEnvVars(svcCtx *serviceContext, globalNamePrefix string) (map[s
 		prefixes = append(prefixes, svcCtx.service.GroupVersionKind().Kind)
 	}
 
-	return envvars.Build(svcCtx.envVars, prefixes...)
+	if svcCtx.namingStrategy == nil {
+		return envvars.Build(svcCtx.envVars, prefixes...)
```

Comment:
A default implementation for `namingStrategy` should be included to behave as `envvars.Build`, so this (and the rest of `envvars` package) can be removed.

Reply:
hey @isutton I am thinking in line that `envvars` package will still prepare list of key,value pairs of env variables, which would go into the intermediate secret. Once we have list of env vars `naming` package would apply naming based on the strategy provided.

Reply:
In the PRs description there's the following: `namingStrategy: '{{ .service.kind | lower }}_custom_svc_{{ .name | lower }}'.

What you're stating is that I'd have an environment variable then called `service_custom_svc_foo_STATUS_LISTENERS_0_TYPE` environment variable at the end (considering the existing `envvars.Build` documentation and, of course, assuming `foo` contains `.status.listeners.0.type`).

I'm not sure this would be the end result I'd expect, but I might be uninformed; could you clarify?

Reply:
@isutton yes this is expected, It would be bit tricky to get custom naming to work with nested types or array. So just want to make sure naming works for `string`, `int`, `float` first. cc: @pedjak